test/e2e/config/azure-dev.yaml

```diff
@@ -198,6 +209,9 @@ variables:
   SECURITY_SCAN_FAIL_THRESHOLD: "${SECURITY_SCAN_FAIL_THRESHOLD:-100}"
   SECURITY_SCAN_CONTAINER: "${SECURITY_SCAN_CONTAINER:-quay.io/armosec/kubescape:v2.0.167}"
   AZURE_CNI_V1_MANIFEST_PATH: "${PWD}/templates/addons/azure-cni-v1.yaml"
+  CAPI_VERSION: "v1.5.4"
```

Comment:
Would it be worth making separate variables for the CAPI version for the old and latest CAPZ versions? Eventually those will drift out of sync and this may avoid some confusion when that happens.

And it may be worth naming this to indicate it's only used in the upgrade tests.